src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1895,7 +1930,10 @@ bool QgsOgrProvider::addFeaturePrivate( QgsFeature &f, Flags flags, QgsFeatureId
 
       if ( !ok )
       {
-        pushError( tr( "wrong data type for attribute %1 of feature %2: %3" ).arg( qgisAttributeId ) .arg( f.id() ).arg( qType ) );
+        if ( !errorEmitted )
+        {
+          pushError( tr( "wrong data type for attribute %1 of feature %2: %3" ).arg( qgisAttributeId ) .arg( f.id() ).arg( qType ) );
```

Comment:
Wouldn't it be better to display the type name instead of the QVariant::type enum value?

Reply:
> Wouldn't it be better to display the type name instead of the QVariant::type enum value?

agreed. I actually tried using QMetaType::name() method (https://doc.qt.io/qt-6/qmetatype.html#name), but it returned a "-" string for 10=String. Probably a mis-use of the Qt API. Ah, actually I see https://doc.qt.io/qt-6/qvariant.html#typeName . Just used that, and it works OK. Commit amended.